samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -274,7 +276,8 @@ public void onContainerFailed(Throwable t) {
               } else {
                 LOGGER.warn("JobCoordinatorLatch was null. It is possible for some component to be waiting.");
               }
-              LOGGER.error("Container failed. Stopping the processor.", t);
+              containerException = t;
+              LOGGER.error("Container failed. Stopping the processor.", containerException);
               container = null;
               stop();
```

Comment:
Another problem, that this will cause the stop() to run in the a different thread (container thread).
May be we should invoke the stop from JobCoordinator on a debounce Thread?